### ISSUE TITLE
printf: support the file type - similar to "%y" in /usr/bin/find

### ISSUE THREAD

**Issue by hi2u** [rhash/RHash#92]
With the `/usr/bin/find` command, and it's own `-printf` fields, you can display the filetype using the `%y` variable.  e.g:

    /usr/bin/find . -printf '%y %p\n'

Would display:

```
f ./this-is-a-regular-file.txt
l ./this-is-a-symlink.txt
s ./this-is-a-socket
d ./a-directory
```

I guess only "f" (regular file) and "l" (symlink) are relevant for rhash, but it would be great if these could be displayed using rhash's own printf output.

Currently I'm combining find + rhash to generate large file lists including checksums.  But this means `find` needs to execute a new rhash process for every file.  If rhash supported the "type" printf field natively, then I could drop the usage of find altogether, and generate these lists using `rhash -r` alone in a single process.

It would be amazing if rhash could support all the fields that `find` does, but obviously that's a lot of extra work.   I think the "type" could be quite handy though for people that want to know which files were originally symlinks -vs- regular files.

**Comment by hi2u** [comment]
Just thinking more about the all-the-other-find-fields thing... perhaps there is / could be another way to use combined the output of both `find` and `rhash` so that you can get the best of both programs, but only using 2 processes?  

This would also allow you to use all the filtering options of find.

I know there's the `--file-list` option in rhash now... maybe there could be a similar option, where a single rhash process takes the piped output of `find` (which may contain many fields, not just the filepaths) and then rhash uses that as the source file list, and modifies its own output to also include the additional checksum fields needed?

Or maybe there's a simpler way to do all of this (without running multiple rhash processes) that doesn't require any new features/changes to rhash or find?

**Comment by rhash** [comment]
I see that the use-case is rare and this feature got no interest from others.

You shall use the following commands instead:
~~~sh
find /usr/bin/ -type l | rhash -p 'l %p\n' --file-list -
find /usr/bin/ -type f | rhash -p 'f %p\n' --file-list -
~~~

Passing stdin (`-') as a --file-list argument was broken since v1.3.7 and it got fixed by d10d941db89d919f3963f8dc9c0113b3d0e40854. Until the fix gets into release you should use a temporary file as a file-list.
